src/sidebar/components/filter-status.js

```diff
@@ -0,0 +1,153 @@
+import { createElement } from 'preact';
+
+import { countVisible } from '../util/thread';
+
+import Button from './Button';
+
+import useRootThread from './hooks/use-root-thread';
+import useStore from '../store/use-store';
+
+/**
+ * Render information about the currently-applied filters and allow the
+ * user to clear filters or toggle user-focus mode on and off
+ *
+ * When any filtering is applied, we are in one of 3 mutually-exclusive filtering
+ * "modes". In any mode, we display descriptive text about what the user
+ * is looking at, and present one button to take action on the current filtering
+ * state:
+ * - selection: One or more annotations have been selected in the document by
+ *   the user, or we're looking at a direct-linked annotation. Button clears
+ *   the selection. This mode supersedes all others (i.e. other filters ignored).
+ * - query: There is a user-entered filter query. It can "stack" with user-focus.
+ *    Button clears the selection (query, etc.), but will leave user-focus mode
+ *    intact.
+ * - focus: Neither of the above two modes apply and user-focus mode is
+ *   configured (it may or may not be active). Button toggles activation of
+ *   user-focus mode.
+ *
+ * Descriptive text about the filter status and annotations follows the
+ * pattern:
+ *
+ *   [Showing] (<resultCount>|No) (annotation[s]|result[s]) [for <filterQuery>]
+ *   [by <focusedUser>] [\(and <forcedCount> more\)]`
```

Comment:
This comment is super helpful. I understand your point about i18n challenges here. I think we can cross that bridge when we get to it.